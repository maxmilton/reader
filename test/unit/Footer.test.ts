// FIXME: Uncomment this file once bun macro issues are fixed:
//  ↳ https://github.com/oven-sh/bun/issues/3641
//  ↳ https://github.com/oven-sh/bun/issues/3832

// import { afterEach, expect, test } from 'bun:test';
// import { Footer } from '../../src/components/Footer';
// import { cleanup, render } from './utils';

// afterEach(cleanup);

// test('rendered DOM contains expected elements', () => {
//   expect.assertions(3);
//   const rendered = render(Footer());
//   expect(rendered.container.firstChild?.nodeName).toBe('FOOTER');
//   expect(rendered.container.querySelector('a[href="https://maxmilton.com"]')).toBeTruthy();
//   expect(
//     rendered.container.querySelector('a[href="https://github.com/maxmilton/reader/issues"]'),
//   ).toBeTruthy();

//   // TODO: More/better assertions
// });

// test('rendered DOM matches snapshot', () => {
//   expect.assertions(1);
//   const rendered = render(Footer());
//   expect(rendered.container.innerHTML).toMatchSnapshot();
// });

// test('contains the app release version number', () => {
//   expect.assertions(1);
//   const rendered = render(Footer());
//   expect(rendered.container.innerHTML).toMatch(/v\d+\.\d+\.\d+/);
// });

$console.warn('FIXME: Footer tests are disabled');
