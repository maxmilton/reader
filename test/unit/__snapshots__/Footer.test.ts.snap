// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
"<footer class="mv2 muted fss tc">
© <a href="https://maxmilton.com" class="normal muted" target="_blank">Max Milton</a> ・ vundefined ・ <a href="https://github.com/maxmilton/reader/issues" target="_blank">report bug</a>
</footer>"
`;
